### DISCUSSION TITLE
Is there a way to set layer color in a viewport?

### DISCUSSION THREAD

**Discussion by kurt-rhee** [mozman/ezdxf#670] [Q&A]
Hello, 

I was wondering if it is possible within a viewport to change a layer color, while leaving modelspace as it is.

Thank you.

**Comment by mozman** [comment]
I added this feature to v0.18b3 and an example how to use it: [viewports_override_layer_attributes.py](https://github.com/mozman/ezdxf/blob/master/examples/viewports_override_layer_attributes.py)

The same lines in different viewports:
1. orignal (modelspace)
2. overriding ACI
3. overriding RGB
4. overriding linetype
5. overriding lineweight

![image](https://user-images.githubusercontent.com/4230966/162680080-2d775b3c-ea0c-4ccb-b405-7ebbcbdb033f.png)

The DXF format support overriding some layer attributes individually for each `VIEWPORT` entity. 
The state of this attributes is  stored in the `LAYER` entity:

- ACI color
- RGB color
- transparency
- linetype
- lineweight

The frozen state of the layer is stored in the `VIEWPORT` entity. I added some management methods in v0.18b3 to the `VIEWPORT` entity:

- `freeze(layer_name)`
- `thaw(layer_name)`
- `is_frozen(layer_name)`
